ion/src/device/bootloader/usb/stack/descriptor/descriptor.h

```diff
@@ -0,0 +1,32 @@
+#ifndef ION_DEVICE_SHARED_USB_STACK_DESCRIPTOR_H
+#define ION_DEVICE_SHARED_USB_STACK_DESCRIPTOR_H
+
+#include "../streamable.h"
+
+namespace Ion {
+namespace Device {
+namespace USB {
+
+class InterfaceDescriptor;
+
+class Descriptor : public Streamable {
+  friend class InterfaceDescriptor;
+public:
+  constexpr Descriptor(uint8_t bDescriptorType) :
+    m_bDescriptorType(bDescriptorType)
+  {
+  }
+  uint8_t type() const { return m_bDescriptorType; }
+protected:
+  void push(Channel * c) const override;
+  virtual uint8_t bLength() const { return 2*sizeof(uint8_t); }
```

Comment:
What means this name ?

Reply:
I don't know it's the same usb code as in the shared directory.